{{cookiecutter.project_name}}/.github/workflows/{% if cookiecutter.__type=='compiled' %}cd.yml{% endif %}

```diff
@@ -30,6 +30,51 @@ jobs:
           name: cibw-sdist
           path: dist/*.tar.gz
 
+  test_sdist:
+    name: Test SDist with python ${{ matrix.python }}
+    needs: [make_sdist]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ["3.8", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        name: Install Python ${{ matrix.python }}
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install dependencies
+        run: |
+          pip install pytest pytest-cov
```

Comment:
The dependencies could be automatically extracted using approach like the one suggested in https://github.com/pypa/pip/issues/8049#issuecomment-788784066 or we could wait for https://github.com/pypa/pip/issues/11440 to be integrated.

Reply:
Here is what I am using in an other project:

```
          # Waiting pip supports `--only-deps=test`, explicitly extract the test dependencies
          # See https://github.com/pypa/pip/issues/11440
          pip install yq
          tomlq -r '.project."optional-dependencies".test[]' pyproject.toml | xargs -d '\n' pip install
```